core/interfaces/src/application.rs

```diff
@@ -79,139 +89,236 @@ pub trait ApplicationInterface<C: Collection>:
+        }
+    }
 
     /// Returns last executed block hash. [0;32] is genesis
-    fn get_last_block(&self) -> [u8; 32];
-
-    /// Returns an accounts current nonce
-    fn get_account_nonce(&self, public_key: &EthAddress) -> u64;
-
-    /// Returns the chain id
-    fn get_chain_id(&self) -> u32;
+    fn get_last_block(&self) -> [u8; 32] {
+        match self.get_metadata(&Metadata::LastBlockHash) {
+            Some(Value::Hash(hash)) => hash,
+            _ => [0; 32],
+        }
+    }
 
-    /// Returns the current block number
-    fn get_block_number(&self) -> u64;
+    /// Returns a full copy of the entire node-registry,
+    /// Paging Params - filtering nodes that are still a valid node and have enough stake; Takes
+    /// from starting index and specified amount.
+    fn get_node_registry(&self, paging: Option<PagingParams>) -> Vec<NodeInfoWithIndex> {
+        let staking_amount = self.get_staking_amount().into();
+
+        self.get_node_table_iter::<Vec<NodeInfoWithIndex>>(|nodes| -> Vec<NodeInfoWithIndex> {
+            let nodes = nodes.map(|index| NodeInfoWithIndex {
+                index,
+                info: self.get_node_info::<NodeInfo>(&index, |n| n).unwrap(),
+            });
+            match paging {
+                None => nodes
+                    .filter(|node| node.info.stake.staked >= staking_amount)
+                    .collect(),
+                Some(PagingParams {
+                    ignore_stake,
+                    limit,
+                    start,
+                }) => {
+                    let mut nodes = nodes
+                        .filter(|node| ignore_stake || node.info.stake.staked >= staking_amount)
+                        .collect::<Vec<NodeInfoWithIndex>>();
+
+                    nodes.sort_by_key(|info| info.index);
+
+                    nodes
+                        .into_iter()
+                        .filter(|info| info.index >= start)
+                        .take(limit)
+                        .collect()
+                },
+            }
+        })
+    }
 
-    /// Checks if an transaction digest has been executed this epoch.
-    fn has_executed_digest(&self, digest: TxHash) -> bool;
+    /// Returns the amount that is required to be a valid node in the network.
+    fn get_staking_amount(&self) -> u128 {
+        self.get_protocol_param(&ProtocolParams::MinimumNodeStake)
+            .unwrap_or(0)
+    }
 
-    /// Returns the uptime for a node from the past epoch.
-    fn get_node_uptime(&self, node_index: &NodeIndex) -> Option<u8>;
+    /// Returns true if the node is a valid node in the network, with enough stake.
+    fn is_valid_node(&self, id: &NodePublicKey) -> bool {
+        let minimum_stake_amount = self.get_staking_amount().into();
+        self.pubkey_to_index(id).is_some_and(|node_idx| {
+            self.get_node_info(&node_idx, |n| n.stake.staked)
+                .is_some_and(|node_stake| node_stake >= minimum_stake_amount)
+        })
+    }
 }
 
```

Comment:
And add the blanket implementation here:

```suggestion
impl<T: SyncQueryRunnerInterface> QueryRunnerExt for T {}
```